clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -611,6 +612,11 @@ private ListOffsetResult fetchOffsetsByTimes(Map<TopicPartition, Long> timestamp
                     if (records.notInitialized()) {
                         try {
                             nextInLineFetch = initializeCompletedFetch(records);
+
+                            // nextInLineRecords might be null when completedFetch contains error
+                            if (nextInLineFetch == null) {
+                                records.response.decRefCount();
```

Comment:
It seems this decRefCount() call represents the removing of the `CompletedFetch` object from the `completedFetches` field. If so, can we move the call after the `completedFetches.poll()` call on line 636 when it's actually removed?